### PR TITLE
[teammgrd]: Add retry logic for starting port channel with teamd

### PR DESCRIPTION
**Pull request by stcheng**
teamd command may fail when failed to allocate memory to create the port
channel. Retry logic make sure the creation will be successful.

Signed-off-by: Shu0T1an ChenG <shuche@microsoft.com>

### CODE REVIEW

**Comment by jipanyang** [comment]
Why memory allocation failed? Physical memory had been exhausted or some internal processing failure within teamd?

**Comment by stcheng** [comment]
retest this please

**Comment by stcheng** [comment]
@jipanyang when multiple teamd instances start too quickly I observe this issue

I have written a very simple script to reproduce this:
```
#!/bin/bash

for i in $(seq 1 32); do
  /usr/bin/teamd -r -t PortChannel00$i -c '{"device":"PortChannel8","hwaddr":"d8:9e:f3:d5:cd:e0","runner":{"active":"true","name":"lacp","min_ports":2}}' -L /var/warmboot/teamd/ -g -d
done
```
And I could see logs:
```
...
Using team device "PortChannel0022".
Using PID file "/var/run/teamd/PortChannel0022.pid"
This program is not intended to be run as root.
Daemon process failed.
Failed: Cannot allocate memory
Using team device "PortChannel0023".
Using PID file "/var/run/teamd/PortChannel0023.pid"
This program is not intended to be run as root.
Daemon process failed.
Failed: Cannot allocate memory
Using team device "PortChannel0024".
Using PID file "/var/run/teamd/PortChannel0024.pid"
This program is not intended to be run as root.
Using team device "PortChannel0025".
Using PID file "/var/run/teamd/PortChannel0025.pid"
This program is not intended to be run as root.
...
```

File: cfgmgr/teammgr.cpp [hunk 1]
```diff
@@ -400,12 +403,18 @@ bool TeamMgr::addLag(const string &alias, int min_links, bool fallback)
         << " -t " << alias
         << " -c " << conf.str()
         << " -L " << dump_path
-        << " -d";
-    EXEC_WITH_ERROR_THROW(cmd.str(), res);
+        << " -g -d";
```
Comment (jipanyang) [inline cfgmgr/teammgr.cpp]: Does -g give give proper log Verbosity Level?

Comment (stcheng) [reply cfgmgr/teammgr.cpp]: yes the debug logs will be in teamd.log file under `/var/log` folder

**Comment by twtseng-tim** [comment]
@stcheng @jipanyang Hello
I tried to create a lot of portchannels, PortChannel0000 to PortChannel1000.
To save config and then reload config, still see the following warning message.
> sonic INFO teamd#supervisord: teammgrd Daemon process failed.
> sonic INFO teamd#supervisord: teammgrd Failed: Cannot allocate memory
The retry mechanism does not work.
The teamd don't add any portchaneel successfully after getting waring message.
Do you have any suggestions for this situation
Thanks.

**Pull request merged**